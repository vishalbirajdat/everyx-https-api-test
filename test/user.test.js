const request = require("supertest");
require('dotenv').config();
const api = request(process.env.API_BASE_URL || "http://localhost:8800");

const normalizeResponse = (response) => ({
  status: response.status,
  body: {
    ...response.body,
    timestamp: "<DYNAMIC>",
    sessionId: response.body.sessionId ? "<DYNAMIC_SESSION_ID>" : undefined
  }
});

describe("Login Flow Snapshot Testing", () => {
  it("POST /login with correct username & password", async () => {
    const response = await api.post("/login").send({ username: "testuser", password: "123456" });
    expect(normalizeResponse(response)).toMatchSnapshot();
  });

  it("POST /login with wrong password", async () => {
    const response = await api.post("/login").send({ username: "testuser", password: "wrongpassword" });
    expect(normalizeResponse(response)).toMatchSnapshot();
  });

  it("POST /login with missing password", async () => {
    const response = await api.post("/login").send({ username: "testuser" });
    expect(normalizeResponse(response)).toMatchSnapshot();
  });

  it("POST /login with missing username", async () => {
    const response = await api.post("/login").send({ password: "123456" });
    expect(normalizeResponse(response)).toMatchSnapshot();
  });
});
