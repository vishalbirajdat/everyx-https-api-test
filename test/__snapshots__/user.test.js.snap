// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Login Flow Snapshot Testing POST /login with correct username & password 1`] = `
{
  "body": {
    "sessionId": undefined,
    "timestamp": "<DYNAMIC>",
  },
  "status": 404,
}
`;

exports[`Login Flow Snapshot Testing POST /login with missing password 1`] = `
{
  "body": {
    "sessionId": undefined,
    "timestamp": "<DYNAMIC>",
  },
  "status": 404,
}
`;

exports[`Login Flow Snapshot Testing POST /login with missing username 1`] = `
{
  "body": {
    "sessionId": undefined,
    "timestamp": "<DYNAMIC>",
  },
  "status": 404,
}
`;

exports[`Login Flow Snapshot Testing POST /login with wrong password 1`] = `
{
  "body": {
    "sessionId": undefined,
    "timestamp": "<DYNAMIC>",
  },
  "status": 404,
}
`;
